src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -38,20 +38,23 @@ public OtpUserController(String apiPrefix) {
 
     @Override
     OtpUser preCreateHook(OtpUser user, Request req) {
-        // Check API key and assign user to appropriate third-party application. Note: this is only relevant for
-        // instances of otp-middleware running behind API Gateway.
         String apiKey = req.headers("x-api-key");
-        ApiUser apiUser = Persistence.apiUsers.getOneFiltered(Filters.eq("apiKeys.value", apiKey));
-        if (apiUser != null) {
-            // If API user found, assign to new OTP user.
-            user.applicationId = apiUser.id;
-        } else {
-            // If API user not found, report to Bugsnag for further investigation.
-            BugsnagReporter.reportErrorToBugsnag(
-                "OTP user created with API key that is not linked to any API user",
-                apiKey,
-                new IllegalArgumentException("API key not linked to API user.")
-            );
+        // If an api key is present an API user is attempting to create an OTP user.
+        if (apiKey != null) {
+            // Check API key and assign user to appropriate third-party application. Note: this is only relevant for
+            // instances of otp-middleware running behind API Gateway.
+            ApiUser apiUser = Persistence.apiUsers.getOneFiltered(Filters.eq("apiKeys.value", apiKey));
```

Comment:
@br648, this block of code kind of speaks to my other comments about first vs. third party user. In otp-admin-ui we want to be able to associate the MOD UI with an ApiUser so that we can see associate requests being made through that application. Accordingly, the idea of an OtpUser being a first party user is somewhat correct, but I think the meaning of firstPartyUser is muddied because in my view it has been intended to mean an ApiUser associated with MOD UI (AKA a first party application).

Reply:
Can we discuss as I'm not sure what needs to change here.